lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -0,0 +1,186 @@
+            desc = "report a #{@expected_error} error"
+          when Exception
+            desc = "report a #{@expected_error.class} error"
+            desc += " with message '#{@expected_error.message}'" unless @expected_error.message.empty?
+          when Regexp
+            desc = "report an error with message matching #{@expected_error}"
+          when Symbol
+            desc = "report #{@expected_error}"
+          end
+          desc += " with #{@attributes}" unless @attributes.empty?
+          desc
+        end
+
+        def failure_message
+          if !@error_subscriber.events.empty? && !@attributes.empty?
+            event_context = @error_subscriber.events.last[1].with_indifferent_access["context"]
+            unmatched = unmatched_attributes(event_context)
+            unless unmatched.empty?
+              return "Expected error attributes to match #{@attributes}, but got these mismatches: #{unmatched} and actual values are #{event_context}"
+            end
+          elsif @error_subscriber.events.empty?
+            return 'Expected the block to report an error, but none was reported.'
+          else
+            case @expected_error
+            when Class
+              return "Expected error to be an instance of #{@expected_error}, but got #{actual_error.class} with message: '#{actual_error.message}'"
+            when Exception
+              return "Expected error to be #{@expected_error.class} with message '#{@expected_error.message}', but got #{actual_error.class} with message: '#{actual_error.message}'"
+            when Regexp
+              return "Expected error message to match #{@expected_error}, but got: '#{actual_error.message}'"
+            when Symbol
+              return "Expected error to be #{@expected_error}, but got: #{actual_error}"
+            else
+              return "Expected specific error, but got #{actual_error.class} with message: '#{actual_error.message}'"
+            end
+          end
+        end
+
+        def failure_message_when_negated
+          error_count = @error_subscriber.events.count
+          error_word = 'error'.pluralize(error_count)
+          verb = error_count == 1 ? 'has' : 'have'
+
+          "Expected the block not to report any errors, but #{error_count} #{error_word} #{verb} been reported."
+        end
+
+        private
+
+        def error_matches_expectation?
+          return true if @expected_error.nil? && @error_subscriber.events.any?
+          return false if actual_error.nil?
+
+          case @expected_error
+          when Class
+            actual_error.is_a?(@expected_error)
+          when Exception
+            actual_error.is_a?(@expected_error.class) &&
+              (@expected_error.message.empty? || actual_error.message == @expected_error.message)
+          when Regexp
+            actual_error.message&.match(@expected_error)
+          when Symbol
+            actual_error == @expected_error
+          else
+            true
```

Comment:
It slipped my mind, what might this other value be?

Reply:
It's defensive programming at it's best :) removed